numpy/ma/tests/test_core.py

```diff
@@ -1808,6 +1808,11 @@ def test_default_fill_value_structured(self):
         assert_equal(f1, expected)
         assert_equal(f2, expected)
 
+    def test_default_fill_value_void(self):
+        dt = np.dtype([('v', 'V7')])
```

Comment:
In general, it's a bit weird how `default_fill_value(dtype).dtype != dtype). On the other hand, that means that the fill_value is preserved when casting to and from smaller integer types.

As an aside, it's remarkable how the void default is the same as the int8 default (`ord('?') == 999999 % 256`) - I can only assume this is deliberate